hazelcast/src/test/java/com/hazelcast/test/IsolatedLoggingRule.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.test;
+
+import com.hazelcast.logging.Logger;
+import com.hazelcast.logging.LoggerFactory;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+import java.lang.reflect.Field;
+
+/**
+ * Fully isolates the shared global logging state for a test. Also provides
+ * utilities which are useful while testing logging related things.
+ * <p>
+ * Make sure you are using a serial test runner if you use this rule.
```

Comment:
I think we can even enforce it. `Description` param of `apply` seems to know about the test and the test class being executed. 

Adding something like the following to the beginning of the `apply` method below may do the trick.
```
        // making sure that there are no concurrent test executions so we don't impact their results unintentionally
        RunWith runWith = description.getTestClass().getAnnotation(RunWith.class);
        if (!HazelcastSerialClassRunner.class.equals(runWith.value())) {
            throw new IllegalStateException("Tests using this rule must use serial runner");
        }
```

Reply:
Good idea, will definitely incorporate this into the fix.